File: 002-Guess-the-Number/Code.java
import java.math.BigInteger;

/*
 * 原题见: https://adworld.xctf.org.cn/task/answer?type=reverse&number=4&grade=1&id=4908&page=1
 * 原题为同目录下的 Guess-the-Number.jar , 从中提取出来的字节码文件为 Guess.class
 */

public class Code {
    static String XOR(String _str_one, String _str_two) 
    {
        BigInteger i1 = new BigInteger(_str_one, 16);
        BigInteger i2 = new BigInteger(_str_two, 16);
        BigInteger res = i1.xor(i2);
        return res.toString(16);
    }

    public static void main(String[] args) 
    {
        String str_one = "4b64ca12ace755516c178f72d05d7061";
        String str_two = "ecd44646cfe5994ebeb35bf922e25dba";
        String answer = XOR(str_one, str_two);
        System.out.println("your flag is: " + answer);
    }
}

File: 025-BABYRE/code.cpp
#include <iostream>

/*
 * 原题见: https://adworld.xctf.org.cn/task/answer?type=reverse&number=4&grade=1&id=4662&page=2
 * 解题步骤见 解题步骤.md
 */

int main()
{
	int flag[14] =
	{
		102, 109, 99, 100, 127,
		107, 55, 100, 59, 86,
		96, 59, 110, 112
	};

	for(int i = 0; i <= 13; ++i)
	{
		std::cout << char(flag[i] ^ i);
	}
}
